### PR TITLE
Add workflows for Black formatting and building GithubPages

### PR DESCRIPTION
**Pull request by cblessing24**
This PR adds two workflows. The first checks the code for Black conformity on pushes and pull requests. The second one builds the documentation and places it in the `gh_pages` branch on pushes to the `master` branch.  Additionally all files have been Black formatted.

### CODE REVIEW

**Review by ArneNx — APPROVED** [review]
I'm not sure how to test the workflow here.
The rest (which only adds missing Black formatting I gather) looks good.

File: nnfabrik/examples/mnist/trainer.py [hunk 1]
```diff
@@ -59,7 +63,7 @@ def mnist_trainer_fn(
     cb: Callable,
     **config
 ) -> Tuple[float, Any, Dict]:
-    """"
+    """ "
```
Comment (ArneNx) [inline nnfabrik/examples/mnist/trainer.py:66]: four quotation marks!

Comment (cblessing24) [reply nnfabrik/examples/mnist/trainer.py]: Nicely spotted. Fixed.

**Comment by cblessing24** [comment]
Yeah testing the workflow isn't easy but it seems to work fine on my fork. I also updated the `README.md` with a badge and a link to the docs just now. Could you merge it if you are happy with it @ArneNx ?

**Pull request merged**